include/boost/json/detail/config.hpp

```diff
@@ -209,6 +211,28 @@
 BOOST_JSON_NS_BEGIN
 namespace detail {
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-noreturn"
+#elif defined(__GNUC__)
+#pragma gcc diagnostic push
+#pragma gcc diagnostic ignored "-Winvalid-noreturn"
+#endif
+
+BOOST_NORETURN
+BOOST_FORCEINLINE
+void
+unreachable()
```

Comment:
doesn't this need to be marked `inline`?

Reply:
`BOOST_FORCEINLINE` does that for us

Reply:
`BOOST_FORCEINLINE` does that for us